exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/OracleDialect.kt

```diff
@@ -37,11 +36,14 @@ internal object OracleDataTypeProvider : DataTypeProvider() {
     override fun booleanFromStringToBoolean(value: String): Boolean = try {
         value.toLong() != 0L
     } catch (ex: NumberFormatException) {
-        error("Unexpected value of type Boolean: $value")
+        try {
+            value.lowercase().toBooleanStrict()
+        } catch (ex: IllegalArgumentException) {
+            error("Unexpected value of type Boolean: $value")
+        }
```

Comment:
I considered refactoring this to something similar to what's done in `vendors/Mysql.kt`:
```kt
override fun booleanFromStringToBoolean(value: String): Boolean = when (value) {
    "0" -> false
    "1" -> true
    else -> value.toBoolean()
}
```
But didn't want to exclude the possibility of DB returning multi-digit strings, like `"000"`. This is the only workaround I could think of that attempts both extensions if 1 throws.

Reply:
Could you tell me in what case it can happen?

Reply:
@e5l To get a multi-digit string? Honestly, I'm not sure, that's just the assumption I'm making for why `value.toLong()` was used instead of the code in `Mysql.kt`.

Reply:
From what I'm reading, boolean datatype in Oracle is [a hot topic](https://asktom.oracle.com/pls/apex/f?p=100:11:0::::P11_QUESTION_ID:6263249199595#876972400346931526) and only just got supported in 23c. But the standard recommendation seems to be 'Y'/'N' (char(1)), 0/1(number), or 'true'/'false'(varchar2), but maybe some people choose "0"/"1".